library/include/hipblas.h

```diff
@@ -184,17 +154,67 @@ typedef struct hipblasDoubleComplex
 #endif
 } hipblasDoubleComplex;
 
+// this isn't needed right now
+// typedef struct hipblasInt8Complex
+// {
+// #ifndef __cplusplus
+
+//     hipblasInt8 x, y;
+
+// #else
+
+// private:
+//     hipblasInt8 x, y;
+
+// public:
+// #if __cplusplus >= 201103L
+//     hipblasInt8Complex() = default;
+// #else
+//     hipblasInt8Complex() {}
+// #endif
+
+//     hipblasInt8Complex(hipblasInt8 r, hipblasInt8 i = 0)
+//         : x(r)
+//         , y(i)
+//     {
+//     }
+
+//     hipblasInt8 real() const
+//     {
+//         return x;
+//     }
+//     hipblasInt8 imag() const
+//     {
+//         return y;
+//     }
+//     void real(hipblasInt8 r)
+//     {
+//         x = r;
+//     }
+//     void imag(hipblasInt8 i)
+//     {
+//         y = i;
+//     }
+
+// #endif
+// } hipblasInt8Complex;
+
+#endif // using internal complex class for API
+
 #if __cplusplus >= 201103L
 #include <type_traits>
 static_assert(std::is_standard_layout<hipblasComplex>{},
               "hipblasComplex is not a standard layout type, and thus is incompatible with C.");
 static_assert(
     std::is_standard_layout<hipblasDoubleComplex>{},
     "hipblasDoubleComplex is not a standard layout type, and thus is incompatible with C.");
-static_assert(std::is_trivial<hipblasComplex>{},
-              "hipblasComplex is not a trivial type, and thus is incompatible with C.");
-static_assert(std::is_trivial<hipblasDoubleComplex>{},
-              "hipblasDoubleComplex is not a trivial type, and thus is incompatible with C.");
+
+// Question: Do we need this check?
+// static_assert(std::is_trivial<hipblasComplex>{},
+//               "hipblasComplex is not a trivial type, and thus is incompatible with C.");
+// static_assert(std::is_trivial<hipblasDoubleComplex>{},
+//               "hipblasDoubleComplex is not a trivial type, and thus is incompatible with C.");
```

Comment:
Does it fail?   It must have passed before no?    The sizeof being 2 * floats etc.  may be equivalent ...

Reply:
It fails with hipComplex types. I'm not sure why we need the check myself [cpp reference](https://en.cppreference.com/w/cpp/types/is_trivial)

Reply:
I think you don't want any default constructor doing anything different than C, for us we are covered by the C++ version  based definitions so maybe just move it into the #else our internal type checking.   Those who opt-into hip complex get what they asked for and the size compatibility is all we want to verify and standard layout.